### PR TITLE
add PfmUserException class

### LINKED ISSUE #406: [PfmExceptions] Create a PfmUserException class
Which would allow to filter exceptions linked to user as "this login already exists" and avoid it to be sent to Sentry.

### PR DESCRIPTION
**Pull request by gaetanHerve**
closes #406

### CODE REVIEW

File: Framework/Router.php [hunk 1]
```diff
@@ -364,13 +364,8 @@ private function createController($urlInfo, Request $request) {
...
      */
     private function manageError(Throwable $exception, $type = '') {
-        $sendToSentry = true;
 
-        if ($exception instanceof PfmAuthException) {
-            $sendToSentry = false;
-        }
-
-        if($sendToSentry && Configuration::get('sentry_dsn', '')) {
+        if($exception->sendReports && Configuration::get('sentry_dsn', '')) {
```
Comment (osallou) [inline Framework/Router.php:368]: nope, depending on exception, sendReports will not exists (php, db, etc. exception)

Comment (gaetanHerve) [reply Framework/Router.php:368]: sendReports exists by inheritance (as property of PfmException) and is overridden when specified in child class. No ?

Comment (osallou) [reply Framework/Router.php:368]: for pfmexceptions yes, not for other exceptions

Comment (gaetanHerve) [reply Framework/Router.php:368]: ah, yeah of course

**Comment by osallou** [comment]
There is merge conflict

**Pull request merged**